lib/findUndefinedIdentifiers.js

```diff
@@ -4,20 +4,20 @@ export default function findUndefinedIdentifiers(ast, globalVariables = []) {
   const result = new Set();
   visitIdentifierNodes(
     ast.program,
-    (
-      {
-        isReference,
-        isJSX,
-        name,
-        definedInScope,
-      },
-    ) => {
+    ({ isReference, isJSX, name, definedInScope }) => {
       if (isJSX && !definedInScope.has('React')) {
         result.add('React'); // Implicit dependency
       }
-      if (!isReference && !definedInScope.has(name)) {
-        result.add(name);
+      if (isReference) {
+        return;
       }
+      if (definedInScope.has(name)) {
+        return;
+      }
+      if (isJSX && /^[a-z]/.test(name)) {
```

Comment:
It might be worth moving the regex up to a constant at the module level, so it doesn't have to be created every time.